core/init.c

```diff
@@ -36,6 +36,14 @@
 #include <auto_init.h>
 #endif
 
+#ifndef CONFIG_BOOT_MSG
+#define CONFIG_BOOT_MSG 1
+#endif
```

Comment:
In order to later be able to model this in Kconfig (during transition phase) it would be better to invert the logic of this switch and do something like:
```C
#ifdef DOXYGEN
#define CONFIG_SKIP_BOOT_MSG 0
#endif
```